# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: f25dfc157ef7
# Full Hash: f25dfc157ef7d4e336f7625fea7586f19d20bc17
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-11 10:49:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add QuotaManager::PrincipalMetadataToPrincipalInfo method; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195570
# ==============================================================================

diff -r 30841c08def5 -r f25dfc157ef7 dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Thu Oct 10 23:28:44 2024 +0000
+++ b/dom/quota/QuotaManager.h	Fri Oct 11 00:06:08 2024 +0000
@@ -586,6 +586,9 @@
   Result<PrincipalMetadata, nsresult> GetInfoFromValidatedPrincipalInfo(
       const PrincipalInfo& aPrincipalInfo);
 
+  static Result<PrincipalInfo, nsresult> PrincipalMetadataToPrincipalInfo(
+      const PrincipalMetadata& aPrincipalMetadata);
+
   static nsAutoCString GetOriginFromValidatedPrincipalInfo(
       const PrincipalInfo& aPrincipalInfo);
 
